frontend/containers/DocumentEditToolbar/DocumentEditToolbar.jsx

```diff
@@ -137,17 +203,23 @@ class DocumentEditToolbar extends Component {
       .length
     const method = documentId ? 'edit' : 'new'
 
-    // By default, we support these two save modes.
-    let saveOptions = {
-      'Save and create new': this.handleSave.bind(this, 'saveAndCreateNew'),
-      'Save and go back': this.handleSave.bind(this, 'saveAndGoBack')
-    }
+    const savedSaveOptionOnUser = state.user.remote.data[FIELD_PUBLISH_SAVE_OPTIONS_LAST_USED]
 
-    // If we're editing an existing document, we also allow users to duplicate
-    // the document.
-    if (method === 'edit') {
-      saveOptions['Save as duplicate'] = this.handleSave.bind(this, 'saveAsDuplicate')
-    }
+    const savedSaveOption = savedSaveOptionOnUser || availableSaveOptions.default
+
+    const localSaveOptions = filterMap(
```

Comment:
Again, not a huge fan of the usage of `Map` here. All we're doing is finding a match in an array-like structure, and yet we end up with this somewhat cryptic expression. Do you think we could use an array and `Array.prototype.find`?

Reply:
well, moving to using `Array.prototype.filter` as it was anyway, for filtering stuff with

Reply:
That's been pushed